# Zillow Clustering Project Data Acquisition Notebook

In [1]:
# imports
import pandas as pd
import numpy as np
import acquire
import warnings
warnings.filterwarnings('ignore')

In [2]:
# connect to database, perform SQL query, write to csv, and convert to pandas DataFrame
df = acquire.get_zillow_cluster()

In [4]:
# what do we have to start with?
acquire.summarize(df)

----------------------------------------------------
DataFrame Head
   heatingorsystemtypeid  buildingclasstypeid  architecturalstyletypeid  \
0                    NaN                  NaN                       NaN   
1                    NaN                  NaN                       NaN   
2                    NaN                  NaN                       NaN   

   airconditioningtypeid  parcelid  typeconstructiontypeid  storytypeid  \
0                    NaN  14297519                     NaN          NaN   
1                    NaN  17052889                     NaN          NaN   
2                    NaN  14186244                     NaN          NaN   

   propertylandusetypeid       id  basementsqft  ...  censustractandblock  \
0                  261.0  1727539           NaN  ...         6.059063e+13   
1                  261.0  1387261           NaN  ...         6.111001e+13   
2                  261.0    11677           NaN  ...         6.059022e+13   

   logerror  transact

### Data Aquistion Key Findings & Takeaways:
- This dataframe has 77574 rows and 68 columns.
- Target variable: 'logerror'
- There are 11 columns that are dtype "object" that either need to be dropped, or converted to int
- There are numerous columns with mostly missing values that either need to be dropped, or imputed
- There are numerous columns that at first glance appear to be duplicates, or similar enough to either be dropped, or joined
- Also need to drop all propertyusetypeid != 261 (singe family residential)